### ISSUE TITLE
potential null pointer dereference in cthd_parse::parse

### ISSUE THREAD

**Issue by ColinIanKing** [intel/thermal_daemon#190]
In function cthd_parse::parse in src/thd_parse.cpp there is a potential null return from the following:

```
                        tmp_value = (char*) xmlNodeListGetString(doc,
                                        cur_node->xmlChildrenNode, 1);
```

tmp_value  can be potentially null and it is not being null checked. Later on it is being dereferenced:

```
                                info_ptr->type.assign((const char*) tmp_value);
```

..and this will segfault.

**Comment by spandruvada** [comment]
Thanks Colin.
I will fix this and release 1.9 later today.
-Srinivas

**Comment by ColinIanKing** [comment]
Thanks, I''ve got a couple of fixes I'm working on, I'll try and get them resolved in the next hour and do a merge request.
